### DISCUSSION TITLE
Material.Avalonia Page Transitions Missing - How to implement?

### DISCUSSION THREAD

**Discussion by bsampica** [AvaloniaUI/Avalonia#12677] [General]
Hey, just a short one, unless I'm doing something completely wrong...always a big possibility.

When changing from the Fluent Theme, to the Material theme as part of Material.Avalonia, I noticed that my page transitions as part of my TransitioningContentControl stoped playing.  Switching nothing else, and putting the Fluent Theme back, they returned.  

```
    <Application.Styles>
        <!-- <FluentTheme /> -->
        <themes:MaterialTheme PrimaryColor="Blue" SecondaryColor="DeepOrange" />
        <materialIcons:MaterialIconStyles />
        <dialog:DialogHostStyles />
    </Application.Styles>
```

I've read through the Keyframe animations, transitions and page transitions sections of the Avalonia documentation, but I'm wondering if I'm doing this the hard way.  I also attempted to implement the custom transition code in the page transitions example, to see if I could get a custom transition going in the Material.Avalonia theme, but I wasn't able to figure out what an IVisual was, and which library it should come from.

https://docs.avaloniaui.net/docs/next/guides/graphics-and-animation/how-to-create-a-custom-page-transition

Am I doing this the hard way?

**Comment by stevemonaco** [comment]
> I wasn't able to figure out what an IVisual was, and which library it should come from.

`IVisual` was removed from Avalonia 0.10.x as part of many interface removals, now only the concrete `Visual` remains. Try that instead and see what happens with the custom transition. I don't use page transitions, so I can't help on the overall issue. Edit: I've PR'd documentation fixes for this page.

**Reply by bsampica** [reply]
Thanks, I was able to implement the custom transition using that tip.  It required a few other small changes, but nothing difficult.  

It runs when the Fluent Theme is active, but not when the Material.Avalonia theme is active.  This sounds like a problem with their theme, and not Avalonia.  I've cross posted on their discussions, to see what comes out of there.  

Thanks for the help!

**Reply by bsampica** [reply]
This was solved by a developer over in the Material.Avalonia theme project.  Some total wizard-ninjary there.  
https://github.com/AvaloniaCommunity/Material.Avalonia/discussions/285
